### PR TITLE
fix: [CDS-69077]: Updated Create Trigger Permission to EXECUTE_PIPELINE & EDIT_PIPELINE

### PR DESCRIPTION
**Pull request by pankajsingh-harness**
#### Screenshots
Screenshot 2023-06-12 at 11 00 38 AM

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/72-triggers/components/Triggers/ArtifactTrigger/ArtifactTriggerWizard.tsx [hunk 1]
```diff
@@ -981,26 +979,9 @@ const ArtifactTriggerWizard = (props: { children: JSX.Element[]; isSimplifiedYAM
...
-        resourceIdentifier: pipelineIdentifier
-      },
-      permissions: [PermissionIdentifier.EXECUTE_PIPELINE],
-      options: {
-        skipCache: true
-      }
-    },
-    [projectIdentifier, orgIdentifier, accountId, pipelineIdentifier]
-  )
+  const isTriggerCreatePermission = useIsTriggerCreatePermission()
```
Comment (pankajsingh-harness) [inline src/modules/72-triggers/components/Triggers/ArtifactTrigger/ArtifactTriggerWizard.tsx:982]: This code has been moved to a hook `useIsTriggerCreatePermission` and used  at all the places. This is the only change in these files.

File: src/modules/72-triggers/components/Triggers/useIsTriggerCreatePermission.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import type { PipelinePathProps } from '@common/interfaces/RouteInterfaces'
+import { usePermission } from '@rbac/hooks/usePermission'
+import { PermissionIdentifier } from '@rbac/interfaces/PermissionIdentifier'
+import { ResourceType } from '@rbac/interfaces/ResourceType'
+
+export const useIsTriggerCreatePermission = (): boolean => {
+  const { pipelineIdentifier } = useParams<PipelinePathProps>()
+  const [isTriggerCreatePermission, setIsTriggerCreatePermission] = useState(false)
+  const [isExecutePipeline, isEditPipeline] = usePermission(
+    {
```
Comment (pankajsingh-harness) [inline src/modules/72-triggers/components/Triggers/useIsTriggerCreatePermission.tsx:19]: No passing `resourceScope` as it is already used in the component as `defaultScope `
```
const { accountId: accountIdentifier, orgIdentifier, projectIdentifier } = useParams<ProjectPathProps>()
const defaultScope = { accountIdentifier, orgIdentifier, projectIdentifier }
```

**Review by sonaligoyal-harness — DISMISSED** [review]

**Comment by pankajsingh-harness** [comment]
Jest coverage is failing with:
```
Jest: "global" coverage threshold for statements (65%) not met: 39.88%
Jest: "global" coverage threshold for branches (50%) not met: 22.12%
Jest: "global" coverage threshold for lines (65%) not met: 39.46%
Jest: "global" coverage threshold for functions (50%) not met: 34.94%
```
Screenshot 2023-06-12 at 11 02 32 AM

**Review by rameshx — DISMISSED** [review]

**Review by ishantmehtaharness — APPROVED** [review]
looks good Pankaj

**Review by rameshx — APPROVED** [review]

**Review by sonaligoyal-harness — APPROVED** [review]

**Pull request merged**